test/e2e/benchmark/throughput.go

```diff
@@ -50,56 +50,23 @@ func E2EThroughput() error {
 		TestDuration:       30 * time.Second,
 		TxClients:          2,
 	}
-	// create a new testnet
-	testNet, err := testnet.New("E2EThroughput", seed,
-		testnet.GetGrafanaInfoFromEnvVar(), manifest.ChainID,
-		manifest.GetGenesisModifiers()...)
-	testnet.NoError("failed to create testnet", err)
 
-	testNet.SetConsensusParams(manifest.GetConsensusParams())
+	benchTest, err := NewBenchmarkTest("E2EThroughput", &manifest)
+	testnet.NoError("failed to create bench test", err)
+
 	defer func() {
 		log.Print("Cleaning up testnet")
-		testNet.Cleanup()
+		benchTest.Cleanup()
 	}()
 
-	testnet.NoError("failed to create genesis nodes",
-		testNet.CreateGenesisNodes(manifest.Validators,
-			manifest.CelestiaAppVersion, manifest.SelfDelegation,
-			manifest.UpgradeHeight, manifest.ValidatorResource))
-
-	// obtain the GRPC endpoints of the validators
-	gRPCEndpoints, err := testNet.RemoteGRPCEndpoints()
-	testnet.NoError("failed to get validators GRPC endpoints", err)
-	log.Println("validators GRPC endpoints", gRPCEndpoints)
-
-	// create tx clients and point them to the validators
-	log.Println("Creating tx clients")
-
-	err = testNet.CreateTxClients(manifest.TxClientVersion, manifest.BlobSequences,
-		manifest.BlobSizes,
-		manifest.TxClientsResource, gRPCEndpoints[:manifest.TxClients])
-	testnet.NoError("failed to create tx clients", err)
+	testnet.NoError("failed to setup nodes", benchTest.SetupNodes())
 
-	// start the testnet
-	log.Println("Setting up testnet")
-	testnet.NoError("failed to setup testnet", testNet.Setup(
-		testnet.WithPerPeerBandwidth(manifest.PerPeerBandwidth),
-		testnet.WithTimeoutPropose(manifest.TimeoutPropose),
-		testnet.WithTimeoutCommit(manifest.TimeoutCommit),
-		testnet.WithPrometheus(manifest.Prometheus),
-	))
-	log.Println("Starting testnet")
-	testnet.NoError("failed to start testnet", testNet.Start())
-
-	// once the testnet is up, start the tx clients
-	log.Println("Starting tx clients")
-	testnet.NoError("failed to start tx clients", testNet.StartTxClients())
-
-	// wait some time for the tx clients to submit transactions
-	time.Sleep(manifest.TestDuration)
+	testnet.NoError("failed to run the bench test", benchTest.Run())
```

Comment:
[nit] bench => benchmark in all of these lines

Reply:
Sure, done https://github.com/celestiaorg/celestia-app/pull/3395/commits/62a8781f9c4db99c45247724d076e6a5d4a4485e